src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1340,6 +1340,24 @@ public void SetBadRequestState(BadHttpRequestException ex)
             _keepAlive = false;
         }
 
+        // Normally this would have been rejected, but the developer opted into allowing the bad behavior.
+        public void ReportAllowedBadRequest(BadHttpRequestException ex)
+        {
+            Log.ConnectionBadRequest(ConnectionId, ex);
```

Comment:
this is new logging that now occurs even without setting the appcontext switch?

Reply:
This method is only called when the appcontext switch is set. Compare to the SetBadRequestState method above that's called when the appcontext switch is not set. Both write the same log, but the new one avoids marking the request as failed/rejected.